includes/Handlers/Product/Woo_SOR.php

```diff
@@ -225,7 +291,111 @@ public static function update_catalog_variation( CatalogObject $catalog_object,
 		 * @see https://github.com/woocommerce/woocommerce-square/issues/570
 		 */
 		if ( 'variation' === $product->get_type() ) {
-			$variation_data->setName( $product->get_name() );
+			$result                = wc_square()->get_api()->retrieve_options_data();
+			$options_data          = isset( $result[1] ) ? $result[1] : array();
+			$parent_product        = wc_get_product( $product->get_parent_id() );
+			$attributes            = $parent_product->get_attributes();
+			$variation_items       = $product->get_attributes();
+			$variation_item_values = array();
+
+			if ( 1 === count( $attributes ) ) {
```

Comment:
I see that these changes contain multiple "actions" that could be new methods to make reading easier. Basically, every time we have a comment here, the following code could be a new method named after the explanation. But this is optional